### ISSUE TITLE
Error!!! Something wrong with PDBid or path...

### ISSUE THREAD

**Issue by chemgeeklian** [reynoldsk/pySCA#8]
I found an issue related to the MSAsearch function in scaTools.py. I am new to python and just came up with a naive solution. But the output it not consist with that described in the pySCA tutorial.

Firstly I have ggsearch36 installed, which means I can use ggsearch36 in command lines :
```
$ ggsearch36
USAGE
 ggsearch36 [-options] query_file library_file
 ggsearch36 -help for a complete option list

DESCRIPTION
 GGSEARCH performs a global/global database searches
 version: 36.3.8 Jul, 2015

COMMON OPTIONS (options must preceed query_file library_file)
 -s:  scoring matrix;
 -f:  gap-open penalty;
 -g:  gap-extension penalty;
 -S   filter lowercase (seg) residues;
 -b:  high scores reported (limited by -E by default);
 -d:  number of alignments shown (limited by -E by default);
 -I   interactive mode;
```

And in python3, I tried 
./scaProcessMSA.py Inputs/s1Ahalabi_1470_nosnakes.an -s 3TGI -c E -t -n
to do MSA for the S1A family.

It gives error outputs:
```
Trying MSASearch with ggsearch
Trying MSASearch with EMBOSS
Trying MSASearch with BioPython
Error!!! Something wrong with PDBid or path...
```

After debugging for days I found this issue comes from this line in scaTools.py:
`i_0 = [i for i in range(len(hd)) if output.split('\t')[1] in hd[i]]  `

It is that the byte type is not consist with str. I just solved it by modifying it to:
`i_0 = [i for i in range(len(hd)) if output.split(b'\t')[1] in bytes(hd[i],'utf-8')]  `

It works and could give the final MSA outputs, but with 205 positions for S1A instead of 245.

Thank you very much for your help.

**Comment by dxu16** [comment]
I encountered the same problem with python 3.7. However, later I realized that it was probably coded based on python 2. After switching to python 2.7, I was able to execute the original code without any problem.

**Comment by chemgeeklian** [comment]
> I encountered the same problem with python 3.7. However, later I realized that it was probably coded based on python 2. After switching to python 2.7, I was able to execute the original code without any problem.

I do not think it is the right way since python2 will be deprecated in 2020.

**Comment by dxu16** [comment]
> > I encountered the same problem with python 3.7. However, later I realized that it was probably coded based on python 2. After switching to python 2.7, I was able to execute the original code without any problem.
> 
> I do not think it is the right way since python2 will be deprecated in 2020.

That is true. I have also tried using your solution and I think I got the same output as in python 2 in the tutorial. Why do you think that there should be 245 positions in the MSA instead of 205?

**Comment by chemgeeklian** [comment]
> > > I encountered the same problem with python 3.7. However, later I realized that it was probably coded based on python 2. After switching to python 2.7, I was able to execute the original code without any problem.
> > 
> > 
> > I do not think it is the right way since python2 will be deprecated in 2020.
> 
> That is true. I have also tried using your solution and I think I got the same output as in python 2 in the tutorial. Why do you think that there should be 245 positions in the MSA instead of 205?

It's just my misunderstanding. Sorry for that.

**Comment by vapourisedman** [comment]
I am having the same problem now.
vapour@osman:~/pySCA-master$ /usr/local/miniconda2/bin/python2.7 ./scaProcessMSA.py Inputs/s1Ahalabi_1470_nosnakes.an -s 3TGI -c E -t -n >> Outputs/s1A_halabi.log 
/usr/local/miniconda2/lib/python2.7/site-packages/Bio/PDB/StructureBuilder.py:91: PDBConstructionWarning: WARNING: Chain E is discontinuous at line 2569.
  PDBConstructionWarning)
/usr/local/miniconda2/lib/python2.7/site-packages/Bio/PDB/StructureBuilder.py:91: PDBConstructionWarning: WARNING: Chain I is discontinuous at line 2570.
  PDBConstructionWarning)
/usr/local/miniconda2/lib/python2.7/site-packages/Bio/PDB/StructureBuilder.py:91: PDBConstructionWarning: WARNING: Chain E is discontinuous at line 2580.
  PDBConstructionWarning)
/usr/local/miniconda2/lib/python2.7/site-packages/Bio/PDB/StructureBuilder.py:91: PDBConstructionWarning: WARNING: Chain I is discontinuous at line 2758.
  PDBConstructionWarning)
Error!!! Something wrong with PDBid or path...

I guess python 2 api is not supported anymore -- and things needs an update. if there is any solution outthere, ill be glad to hear it.

Cheers,

**Comment by davidkastner** [comment]
I am also getting the same issue for all of the tutorials. I have a new 2020 Macbook pro and have an anaconda environment set up that allows me to work in Python2.7.

Input:
`>> ./scaProcessMSA.py Inputs/PF00186_full.an -s 1RX2 -c A -f 'Escherichia coli' -t -n`

Output:
`Loaded alignment of 5237 sequences, 756 positions.`
`Checking alignment for non-standard amino acids`
`Aligment size after removing sequences with non-standard amino acids: 5228`
`Trimming alignment for highly gapped positions (80% or more).`
`Alignment size post-trimming: 168 positions`
`Finding reference sequence using species-based best match..`
`Trying MSASearch with ggsearch`
`Trying MSASearch with EMBOSS`
`Trying MSASearch with BioPython`
`Cant find the reference sequence using species-based best_match! Using global MSAsearch...`
`Trying MSASearch with ggsearch`
`Trying MSASearch with EMBOSS`
`Trying MSASearch with BioPython`
`Error!!! Something wrong with PDBid or path..`

**Comment by priyankavsetty** [comment]
Finding reference sequence using species-based best match..
Trying MSASearch with ggsearch
Error: [Errno 2] No such file or directory: 'ggsearch36'
Trying MSASearch with EMBOSS
Error: Non-zero return code 127 from 'needle -outfile=tmp_needle_seq.fasta -asequence=tmp_pdb_seq.fasta -bsequence=tmp_algn_seq.fasta -gapopen=10 -gapextend=0.5', message '/bin/sh: 1: needle: not found'
Trying MSASearch with BioPython
Error: both sequences must be of the same type, either string/sequence object or list. Gap character must fit the sequence type (string or list)
Cant find the reference sequence using species-based best_match! Using global MSAsearch...
Trying MSASearch with ggsearch
Error: [Errno 2] No such file or directory: 'ggsearch36'
Trying MSASearch with EMBOSS
Error: Non-zero return code 127 from 'needle -outfile=tmp_needle_seq.fasta -asequence=tmp_pdb_seq.fasta -bsequence=tmp_algn_seq.fasta -gapopen=10 -gapextend=0.5', message '/bin/sh: 1: needle: not found'
Trying MSASearch with BioPython
Error: both sequences must be of the same type, either string/sequence object or list. Gap character must fit the sequence type (string or list)
Error: Error! Can't find reference sequence...
Error!!! Something wrong with PDBid or path...


I have this error with python 3. I cannot find the part of code mentioned earlier by [chemgeeklian](https://github.com/chemgeeklian) in scatools.py anymore. If anyone can help me resolve this it will be great!

Thank you
